### PR TITLE
[cpp-wrapper] fixed issues for generating py args

### PR DESCRIPTION
**Pull request by chenyang78**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #132093

This PR fixed two issues for generating py args:

* in the case of broadcasting, raw_arg can be of int
even if the schema's type is Tensor

* handle cases where any raw_arg is None

This resolved github issue: https://github.com/pytorch/pytorch/issues/131334

After the fix:

```
$ TORCHINDUCTOR_ABI_COMPATIBLE=1 TORCHINDUCTOR_CPP_WRAPPER=1 python test/inductor/test_torchinductor.py CpuTests.test_complex_fallback_cpu
...
Ran 1 test in 3.043s

OK
```

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @yf225 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang

### CODE REVIEW

**Comment by chenyang78** [comment]
xref #131334

File: torch/_inductor/codegen/cpp_wrapper_cpu.py [hunk 1]
```diff
@@ -2255,11 +2261,13 @@ def generate_extern_kernel_alloc_and_find_schema_if_needed_jit(
             self.load_custom_op_wrapper()
 
             assert output_args is not None, "output_args should not be None"
-            num_args = len(raw_args)
+            # raw arg may be None
```
Comment (desertfire) [inline torch/_inductor/codegen/cpp_wrapper_cpu.py:2264]: Do we still need to pass the raw arg if it is None?

Comment (chenyang78) [reply torch/_inductor/codegen/cpp_wrapper_cpu.py:2264]: > Do we still need to pass the raw arg if it is None?

No, we don't. The change above (line 2217 and 2218) skips any None raw arg.

**Review by sijiac — APPROVED** [review]

File: torch/_inductor/codegen/cpp_wrapper_cpu.py (hunk 1 above)
Comment (desertfire) [reply torch/_inductor/codegen/cpp_wrapper_cpu.py:2264]: If we have raw args like `buf0, a=None, b=1`, I think we should still pass in the second parameter as None/Py_None?

**Review by desertfire — COMMENTED** [review]
Also, please add test_complex_fallback_cpu to test_cpu_cpp_wrapper.py

**Pull request closed without merging**